File: java/template-method/src/main/ChildrenScoreCalculator.java
package main;

import java.time.LocalTime;

public class ChildrenScoreCalculator extends BaseScoreCalculator {
    @Override
    public int CalculateOverallScore(int score, int reduction) {
        return score - reduction;
    }

    @Override
    public int CalculateReduction(LocalTime time) {
        return time.getSecond()/2;
    }

    @Override
    public int CalculateBaseScore(int hits) {
        return hits * 80;
    }
}

File: csharp/TemplateMethod/ChildrenScoreCalculator.cs
﻿namespace TemplateMethod;

public class ChildrenScoreCalculator : BaseScoreCalculator
{
    public override int CalculateBaseScore(int hits)
    {
        return hits * 80;
    }

    public override int CalculateOverallScore(int score, int reduction)
    {
        return score - reduction;
    }

    public override int CalculateReduction(TimeSpan time)
    {
        return (int)time.Seconds/2;
    }
}